process/interceptors/singleDataInterceptor_test.go

```diff
@@ -82,14 +83,16 @@ func TestNewSingleDataInterceptor_NilInterceptorThrottlerShouldErr(t *testing.T)
 func TestNewSingleDataInterceptor(t *testing.T) {
 	t.Parallel()
 
+	factory := &mock.InterceptedDataFactoryStub{}
 	sdi, err := interceptors.NewSingleDataInterceptor(
-		&mock.InterceptedDataFactoryStub{},
+		factory,
 		&mock.InterceptorProcessorStub{},
 		&mock.InterceptorThrottlerStub{},
 	)
 
-	assert.NotNil(t, sdi)
-	assert.Nil(t, err)
+	require.False(t, check.IfNil(sdi))
+	require.Nil(t, err)
+	assert.True(t, sdi.InterceptedDataFactory() == factory)
```

Comment:
why not assert.equal?

Reply:
Test removed. The assert.True tested that the same pointer has been set (set worked correctly)